addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerListScreen.js

```diff
@@ -87,7 +87,9 @@ odoo.define('point_of_sale.PartnerListScreen', function(require) {
                 let indexOfSelectedPartner = res.findIndex( partner => 
                     partner.id === this.state.selectedPartner.id
                 );
-                if (indexOfSelectedPartner !== -1) {
+                if (indexOfSelectedPartner === -1) {
+                    res.unshift(this.state.selectedPartner); 
```

Comment:
Maybe instead of this diff, you can keep the original if-statement and move the `res.unshift(this.state.selectedPartner);` below it.